src/model/types.ts

```diff
@@ -63,96 +75,25 @@ export function makeApiProblem(
-  return match<unknown, ApiError>(error)
-    .with({ code: ErrorCode.MissingBearer, message: P.string }, (error) =>
-      makeApiProblem(
-        ErrorCode.MissingBearer,
-        400,
-        error.message,
-        "Bearer token has not been passed"
-      )
-    )
-    .with({ code: ErrorCode.MissingClaim, message: P.string }, (error) =>
-      makeApiProblem(ErrorCode.MissingClaim, 400, error.message, error.message)
-    )
-    .with({ code: ErrorCode.MissingHeader, message: P.string }, (error) =>
-      makeApiProblem(ErrorCode.MissingHeader, 400, error.message, error.message)
-    )
-    .otherwise(() =>
-      makeApiProblem(
-        ErrorCode.MissingHeader,
-        400,
-        "Generic error while processing authorization header",
-        "Unexpected error"
-      )
-    );
-}
+const servicePrefix = "catalog";
 
-export function mapCatalogServiceErrorToApiError(error: unknown): ApiError {
+export function makeApiError(error: unknown): ApiError {
   return match<unknown, ApiError>(error)
-    .with(
-      { code: ErrorCode.DuplicateEserviceName, message: P.string },
-      (error) =>
-        makeApiProblem(
-          ErrorCode.DuplicateEserviceName,
-          409,
-          error.message,
-          "Duplicated service name"
-        )
-    )
-    .with(
-      { code: ErrorCode.ContentTypeParsingError, message: P.string },
-      (error) =>
-        makeApiProblem(
-          ErrorCode.ContentTypeParsingError,
-          400,
-          error.message,
-          "Malformed request"
-        )
-    )
-    .with({ code: ErrorCode.EServiceNotFound, message: P.string }, (error) =>
-      makeApiProblem(
-        ErrorCode.EServiceNotFound,
-        404,
-        error.message,
-        "EService not found"
-      )
-    )
-    .with(
-      { code: ErrorCode.EServiceCannotBeUpdatedOrDeleted, message: P.string },
-      (error) =>
-        makeApiProblem(
-          ErrorCode.EServiceCannotBeUpdatedOrDeleted,
-          400,
-          error.message,
-          "EService cannot be updated"
-        )
-    )
-    .with({ code: ErrorCode.OperationForbidden, message: P.string }, (error) =>
+    .with(P.instanceOf(CatalogProcessError), (error) =>
       makeApiProblem(
-        ErrorCode.OperationForbidden,
-        400,
-        error.message,
-        "Operation forbidden"
+        error.type.code,
+        error.type.httpStatus,
+        error.type.title,
+        error.message
       )
     )
-    .with(
-      { code: ErrorCode.ContentTypeParsingError, message: P.string },
-      (error) =>
-        makeApiProblem(
-          ErrorCode.ContentTypeParsingError,
-          500,
-          error.message,
-          "Internal server error"
-        )
-    )
     .otherwise(() =>
       makeApiProblem(
-        ErrorCode.GenericError,
-        500,
-        "Generic error while processing catalog process error",
-        "Unexpected error"
+        `${servicePrefix}-${ErrorTypes.GenericError.code}`,
```

Comment:
perché vogliamo un prefisso qui? Non lo fa il BFF questo lavoro?

Reply:
attualmente mi sembra che lo facciano i singoli servizi tramite [questo codice](https://github.com/pagopa/interop-commons/blob/1.0.x/utils/src/main/scala/it/pagopa/interop/commons/utils/errors/Problem.scala/#L37) dentro la libreria commons

Reply:
mi torna che venga generato dal servizio stesso, in modo da fornire al chiamante un codice univoco alla fonte.
Oltretutto il BFF non sarà l'unico client, specificarlo qui elimina la definizione duplicata 👍